facefusion/processors/modules/face_editor.py

```diff
@@ -233,29 +219,81 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
 			'target': motion_points,
 			'source': motion_points_edit
 		})[0][0]
-
 	return crop_vision_frame
 
 
-def calc_face_edit_ratio(face_landmark_68 : FaceLandmark68) -> Tuple[NDArray[Any], NDArray[Any]]:
+def edit_eye_gaze(expression : NDArray[Any]) -> NDArray[Any]:
+	face_editor_eye_gaze_horizontal = state_manager.get_item('face_editor_eye_gaze_horizontal')
+	face_editor_eye_gaze_vertical = state_manager.get_item('face_editor_eye_gaze_vertical')
+
+	if face_editor_eye_gaze_horizontal > 0:
+		expression[0, 11, 0] += map_float(face_editor_eye_gaze_horizontal, -1, 1, -0.015, 0.015)
+		expression[0, 15, 0] += map_float(face_editor_eye_gaze_horizontal, -1, 1, -0.020, 0.020)
+	else:
+		expression[0, 11, 0] += map_float(face_editor_eye_gaze_horizontal, -1, 1, -0.020, 0.020)
+		expression[0, 15, 0] += map_float(face_editor_eye_gaze_horizontal, -1, 1, -0.015, 0.015)
+	expression[0, 1, 1] += map_float(face_editor_eye_gaze_vertical, -1, 1, -0.0025, 0.0025)
+	expression[0, 2, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.0025, 0.0025)
+	expression[0, 11, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.010, 0.010)
+	expression[0, 13, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.005, 0.005)
+	expression[0, 15, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.010, 0.010)
+	expression[0, 16, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.005, 0.005)
+	return expression
+
+
+def edit_eye_open(motion_points : NDArray[Any], face_landmark_68 : FaceLandmark68) -> NDArray[Any]:
+	eye_retargeter = get_inference_pool().get('eye_retargeter')
 	face_editor_eye_open_ratio = state_manager.get_item('face_editor_eye_open_ratio')
+	left_eye_ratio = calc_distance_ratio(face_landmark_68, 37, 40, 39, 36)
+	right_eye_ratio = calc_distance_ratio(face_landmark_68, 43, 46, 45, 42)
+
+	if face_editor_eye_open_ratio < 0:
+		close_eye_ratio = numpy.array([ left_eye_ratio, right_eye_ratio, 0 ]).astype(numpy.float32).reshape(1, -1)
+
+		with thread_semaphore():
+			close_eye_motion_points = eye_retargeter.run(None,
+			{
+				'input': numpy.concatenate([ motion_points.reshape(1, -1), close_eye_ratio ], axis = 1)
```

Comment:
do the concatenate outside of the semaphore block: potential better performance (I had similar issues before) but for sure better readability